ex/loqui/lib/loqui/client.ex

```diff
@@ -0,0 +1,358 @@
+defmodule Loqui.Client do
+  defmodule State do
```

Comment:
Seems ok, it's a bit more encapsulated. It sort of feels like we're just reimplementing map there.